File: marcel-lexer/src/main/java/com/tambapps/marcel/lexer/MarcelLexerException.java
package com.tambapps.marcel.lexer;

import lombok.Getter;

@Getter
public class MarcelLexerException extends RuntimeException {

  private final int line;
  private final int column;
  public MarcelLexerException(int line, int column, String message) {
    super(String.format("Lexer error at line %d, column %d: %s", line, column, message));
    this.line = line;
    this.column = column;
  }
}

File: marcel-compilation/marcel-lexer/src/main/kotlin/com/tambapps/marcel/lexer/MarcelLexerException.kt
package com.tambapps.marcel.lexer


/**
 * Marcel lexer exception
 *
 * @property line the line at which the error did occur
 * @property column the column at which the error did occur
 * @constructor
 *
 * @param message the message of the error
 */
class MarcelLexerException constructor(val line: Int, val column: Int, message: String?) :
  RuntimeException(String.format("Lexer error at line %d, column %d: %s", line + 1, column, message))
